### PR TITLE
[sdk-manage] Handle zypper exit codes. Fixes JB#51186

### PR DESCRIPTION
**Pull request by Thaodan**

### CODE REVIEW

File: sdk-setup/src/sdk-manage [hunk 1]
```diff
@@ -509,6 +509,18 @@ object_zypper() {
     local object=$1 args=("${@:2}")
 
     enter "$object" zypper --non-interactive "${args[@]}"
+    case $? in
+        100|101|102)
+            # Ignored exit codes:
+            # 100 - ZYPPER_EXIT_INF_UPDATE_NEEDED
+            # 101 - ZYPPER_EXIT_INF_SEC_UPDATE_NEEDED
+            # 102 - ZYPPER_EXIT_INF_REBOOT_NEEDED
+            :
```
Comment (martyone) [inline sdk-setup/src/sdk-manage:518]: I would consider an explicit `return 0` as a more readable alternative :)

File: sdk-setup/src/sdk-manage [hunk 2]
```diff
@@ -509,6 +509,18 @@ object_zypper() {
...
+            ;;
+        103)
+            # 103 - ZYPPER_EXIT_INF_RESTART_NEEDED
+            object_zypper "$@"
```
Comment (martyone) [inline sdk-setup/src/sdk-manage:522]: Worth to care about endless restart loop?

nitpick: I would prefer not omitting `;;` here.

Comment (Thaodan) [reply sdk-setup/src/sdk-manage:522]: That won't happen unless you push endless zypper updates. After running zypper again the return status is different.

File: sdk-setup/src/sdk-manage (hunk 1 above)
Comment (Thaodan) [reply sdk-setup/src/sdk-manage:518]: Can change that however you prefer it. I agree the meaning of : isn't that obvious.

**Review by martyone — COMMENTED** [review]
Rebase on top of current master please

**Comment by Thaodan** [comment]
done

**Comment by martyone** [comment]
Rebase once again please :)

**Pull request merged**